### PR TITLE
storage: support nested paths in file storage

### PR DESCRIPTION
**Pull request by edvbld**
Hi all,

please review this path that makes `FileStorage` with files in nested directories.

Testing:
- `make test` passes on Linux x64
- Manual testing on Linux x64

Thanks,
Erik

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

### Reviewers
 * Robin Westberg ([rwestberg](@rwestberg) - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/skara pull/760/head:pull/760`
`$ git checkout pull/760`

### CODE REVIEW

File: storage/src/main/java/org/openjdk/skara/storage/FileStorage.java [hunk 1]
```diff
@@ -64,6 +64,10 @@ public void put(Collection<T> items) {
             return;
         }
         try {
+            Files.createDirectories(file.getParent());
+            if (!Files.exists(file)) {
```
Comment (rwestberg) [inline storage/src/main/java/org/openjdk/skara/storage/FileStorage.java:68]: Doesn't hurt, but Files.writeString should create it if it doesn't exist as well I think.

**Review by rwestberg — APPROVED** [review]
Looks good!

File: storage/src/main/java/org/openjdk/skara/storage/FileStorage.java (hunk 1 above)
Comment (edvbld) [reply storage/src/main/java/org/openjdk/skara/storage/FileStorage.java:68]: It didn't for nested paths, so the `Files.createDirectories` is definitely needed. The `if (!Files.exists)` could potentially be left out, but otoh it doesn't hurt to be explicit here.

**Pull request closed without merging**